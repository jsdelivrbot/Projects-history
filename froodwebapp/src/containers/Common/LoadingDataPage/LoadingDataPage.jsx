import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Spin } from 'antd';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import { withRouter } from 'react-router-dom';
import config from 'config';
import { loadingDataPage } from './LoadingDataPage.scss';

const mapStateToProps = state => ({
  user: state.login.user,
  commonDataLoading: state.commonData.commonDataLoading,
  commonDataLoaded: state.commonData.commonDataLoaded,
});

const mapDispatchToProps = {
  push
};

export class LoadingDataPage extends Component {
  componentDidMount() {
    if (!this.props.user) {
      this.props.push('/login');
    }
  }

  componentWillReceiveProps(nextProps) {
    this.handleRedirect(nextProps.commonDataLoaded, nextProps.location);
  }

  handleRedirect = (commonDataLoaded, location) => {
    if (commonDataLoaded) {
      const redirectFrom = location.state && location.state.redirectFrom;
      const redirectToId = location.state && location.state.id;
      const redirectToNewScreen = redirectFrom && redirectFrom.includes('new');

      if (redirectFrom && redirectToNewScreen) {
        this.props.push({
          pathname: redirectFrom,
          state: {
            id: 'new'
          }
        });
      } else if (redirectFrom && redirectToId) {
        this.props.push({
          pathname: redirectFrom,
          state: {
            id: redirectToId
          }
        });
      } else if (redirectFrom) {
        this.props.push(redirectFrom);
      } else {
        this.props.push(config.homeRedirectLink);
      }
    }
  }

  render() {
    const { commonDataLoading } = this.props;
    return (
      <div className={ loadingDataPage }>
        <Spin
          size="large"
          spinning={ commonDataLoading }
          tip="Loading Application Data..."
        />
      </div>
    );
  }
}

LoadingDataPage.propTypes = {
  user: PropTypes.object.isRequired,
  commonDataLoading: PropTypes.bool.isRequired,
  commonDataLoaded: PropTypes.bool.isRequired,
  // router
  push: PropTypes.func.isRequired,
  location: PropTypes.object.isRequired,
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(LoadingDataPage));
