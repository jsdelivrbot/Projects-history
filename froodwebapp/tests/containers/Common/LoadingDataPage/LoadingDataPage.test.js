import { LoadingDataPage } from 'containers/Common/LoadingDataPage/LoadingDataPage';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('LoadingDataPage', () => {
  const props = {
    user: {},
    commonDataLoaded: true,
    commonDataLoading: false,
    push: jest.fn(),
    location: {
      pathname: 'pathname',
      newPath: 'newPath'
    }
  };

  it('renders the LoadingDataPage component', () => {
    const component = shallow(<LoadingDataPage { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of LoadingDataPage', () => {
    const component = shallow(<LoadingDataPage { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });

  it('calls push function in handleRedirect function with correct param', () => {
    const location = {
      state: {
        redirectFrom: 'redirectForm',
        id: '1'
      }
    };
    const component = shallow(<LoadingDataPage { ...props } />);
    component.instance().handleRedirect(props.commonDataLoaded, location);

    expect(props.push.mock.calls.length).toBe(1);

    expect(props.push.mock.calls[0][0]).toEqual({
      pathname: location.state.redirectFrom,
      state: {
        id: location.state.id
      }
    });
  });
});
