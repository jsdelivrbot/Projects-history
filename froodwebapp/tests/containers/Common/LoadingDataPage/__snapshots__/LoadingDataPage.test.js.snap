// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingDataPage capturing Snapshot of LoadingDataPage 1`] = `
<div
  className="loadingDataPage"
>
  <Spin
    prefixCls="ant-spin"
    size="large"
    spinning={false}
    tip="Loading Application Data..."
    wrapperClassName=""
  />
</div>
`;
